### PR TITLE
[docs] Rewrite doc for vcpkg_cmake_configure. Move up into docs/maintainers/.

### PR DESCRIPTION
**Pull request by ras0219-msft**
This is the first docs overhaul PR for a helper function from a port.

A key change I propose for the port-based helpers is to de-segregate them from the "built-in" helpers and instead rely on documentation within the page to guide users on how to consume it. This change accomplishes two goals:
1. Improve discoverability - _All_ helper functions (built-in + curated catalog) will be findable in the same location, so users don't need to look in multiple places.
2. Improve usefulness - By including the usage syntax to depend upon the providing port, users have a "one-stop shop" to answer the common scenario of "I was told to use X, how do I do that?".

### CODE REVIEW

**Review by vicroms — DISMISSED** [review]

File: docs/maintainers/vcpkg_cmake_configure.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+### WINDOWS_USE_MSBUILD
+Use MSBuild instead of [Ninja][ninja] when targeting a Windows platform.
+
+By default vcpkg prefers to use Ninja as the CMake Generator for all platforms. However, there are edge cases where MSBuild has different behavior than Ninja. This flag should only be passed if the project requires MSBuild to build correctly.
+
+### GENERATOR
+Specifies the Generator to use.
+
+This is useful if the project-specific buildsystem has been wrapped in a cmake script that won't perform an actual build. If used for this purpose, it should be set to `"Ninja"`.
```
Comment (LilyWangLL) [inline docs/maintainers/vcpkg_cmake_configure.md:64]:
```suggestion
This is useful if the project-specific buildsystem has been wrapped in a CMake script that won't perform an actual build. If used for this purpose, it should be set to `"Ninja"`.
```

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

File: docs/maintainers/vcpkg_cmake_configure.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+## Examples
+
+```cmake
+vcpkg_cmake_configure(
+    SOURCE_PATH "${source_path}"
+    OPTIONS
+        -DBUILD_EXAMPLES=OFF
+        -DBUILD_TESTS=OFF
+)
```
Comment (ras0219-msft) [inline docs/maintainers/vcpkg_cmake_configure.md:122]:
```suggestion
vcpkg_from_github(OUT_SOURCE_PATH source_path ...)
vcpkg_cmake_configure(
    SOURCE_PATH "${source_path}"
    OPTIONS
        -DBUILD_EXAMPLES=OFF
        -DBUILD_TESTS=OFF
)
vcpkg_cmake_install()
```

File: docs/maintainers/vcpkg_cmake_configure.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+```
+
+To use this function, you must depend on the helper port [`vcpkg-cmake`](ports/vcpkg-cmake.md):
+```json
+{
+    "dependencies": [
+        {"name": "vcpkg-cmake", "host": true}
+    ]
+}
```
Comment (ras0219-msft) [inline docs/maintainers/vcpkg_cmake_configure.md:36]:
```suggestion
```no-highlight
"dependencies": [
  {"name": "vcpkg-cmake", "host": true}
]
```

File: docs/maintainers/portfile-functions.md [hunk 4]
```diff
@@ -71,7 +71,7 @@
 ### [vcpkg-cmake](ports/vcpkg-cmake.md)
 
 - [vcpkg\_cmake\_build](ports/vcpkg-cmake/vcpkg_cmake_build.md)
-- [vcpkg\_cmake\_configure](ports/vcpkg-cmake/vcpkg_cmake_configure.md)
+- [vcpkg\_cmake\_configure](vcpkg_cmake_configure.md)
```
Comment (ras0219-msft) [inline docs/maintainers/portfile-functions.md:74]: Should these be merged into the list above?

File: docs/maintainers/portfile-functions.md [hunk 5]
```diff
@@ -18,7 +18,7 @@
 - [vcpkg\_clean\_executables\_in\_bin](vcpkg_clean_executables_in_bin.md)
 - [vcpkg\_clean\_msbuild](vcpkg_clean_msbuild.md)
 - [vcpkg\_common\_definitions](vcpkg_common_definitions.md)
-- [vcpkg\_configure\_cmake](vcpkg_configure_cmake.md) (deprecated, use [vcpkg\_cmake\_configure](ports/vcpkg-cmake/vcpkg_cmake_configure.md))
+- [vcpkg\_configure\_cmake](vcpkg_configure_cmake.md) (deprecated, use [vcpkg\_cmake\_configure](vcpkg_cmake_configure.md))
```
Comment (ras0219-msft) [inline docs/maintainers/portfile-functions.md:21]: (Not in this PR) Deprecated functions should be extracted into a separate section.

File: docs/maintainers/vcpkg_cmake_configure.md (hunk 3 above)
Comment (dg0yt) [reply docs/maintainers/vcpkg_cmake_configure.md:36]: Please don't make examples which wouldn't pass `vcpkg format-manifest`. It is a trap for new contributors.

File: docs/maintainers/portfile-functions.md (hunk 4 above)
Comment (vicroms) [reply docs/maintainers/portfile-functions.md:74]: In my opinion, yes

File: docs/maintainers/vcpkg_cmake_configure.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+vcpkg_cmake_configure(
+    SOURCE_PATH "${source_path}"
+    OPTIONS
+        -DBUILD_EXAMPLES=OFF
+        -DBUILD_TESTS=OFF
+)
+vcpkg_cmake_install()
+```
+
+* [GitHub Search](https://github.com/microsoft/vcpkg/search?q=vcpkg_cmake_configure+path%3A%2Fports)
```
Comment (vicroms) [inline docs/maintainers/vcpkg_cmake_configure.md:128]:
```suggestion
* [View examples in vcpkg](https://github.com/microsoft/vcpkg/search?q=vcpkg_cmake_configure+path%3A%2Fports)
```

Alternatively: "View examples in repository"

**Review by vicroms — CHANGES_REQUESTED** [review]

**Comment by vicroms** [comment]
LGTM after the minor change I suggested.

**Review by vicroms — APPROVED** [review]

**Pull request merged**